ext/preprocess/FileSystemGlobbingInternalization.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Framework.Runtime;
+
+namespace NuGet.Compiler.Preprocess
+{
+    public class FileSystemGlobbingInternalization : ICompileModule
+    {
+        private readonly string _appBase;
+
+        public FileSystemGlobbingInternalization(IServiceProvider services)
+        {
+            var appEnv = services.GetService(typeof(IApplicationEnvironment)) as IApplicationEnvironment;
+            _appBase = appEnv.ApplicationBasePath;
+        }
+
+        public void BeforeCompile(IBeforeCompileContext context)
+        {
+            var fileSystemGlobbingDir = Path.Combine(_appBase, "..", "..", "submodules", "FileSystem", "src",
```

Comment:
Since Coherence-Signed didn't pass and your change didn't get in, I'll push the current implementation to unblock @anurse for now. I will change it to use more accurate info in another PR.